src/renderer/components/+cluster-settings/components/cluster-metrics-setting.tsx

```diff
@@ -0,0 +1,118 @@
+import "./cluster-metrics-setting.scss";
+
+import React from "react";
+import { observer } from "mobx-react";
+import { Select, SelectOption } from "../../select/select";
+import { Icon } from "../../icon/icon";
+import { Button } from "../../button/button";
+import { SubTitle } from "../../layout/sub-title";
+import { Cluster } from "../../../../main/cluster";
+import { observable } from "mobx";
+
+interface Props {
+  cluster: Cluster;
+}
+
+export enum ResourceType {
+  Cluster = "Cluster",
+  Node = "Node",
+  Pod = "Pod",
+  Deployment = "Deployment",
+  StatefulSet = "StatefulSet",
+  Container = "Container",
+  Ingress = "Ingress",
+  VolumeClaim = "VolumeClaim",
+  ReplicaSet = "ReplicaSet",
+  DaemonSet = "DaemonSet",
+}
+
+@observer
+export class ClusterMetricsSetting extends React.Component<Props> {
+  @observable hiddenMetrics = observable.set<string>();
+
+  componentDidMount() {
+    this.hiddenMetrics = observable.set<string>(this.props.cluster.preferences.hiddenMetrics);
+  }
+
+  save = () => {
+    this.props.cluster.preferences.hiddenMetrics = Array.from(this.hiddenMetrics);
+  };
```

Comment:
Because `preferences` is already an observable. I know that other settings have done it in a similar way but we can make this simpler.
```suggestion
  @computed get hiddenMetrics(): Set<string> {
    return new Set(this.props.cluster.preferences.hiddenMetrics);
  }
  
  componentDidMount() {
    disposeOnUnmount(this, [
      reaction(() => this.hiddenMetrics, newHiddenMetrics => {
        this.props.cluster.preferences.hiddenMetrics = Array.from(newHiddenMetrics);
      }),
    ]);
  }
```

Reply:
@Nokel81 Please reconsider if this new decision will get things simpler. At first glance yes, we eliminate one `save` method, but we also introducing `disposeOnUnmount` here with a circular dependency (as it looks for me at first glance). We react on `this.hiddenMetrics` change and then actually rewrite same value we react on?

Also, not quite clear how we can make some operations like `this.hiddenMetrics.add(value)` on `computed` value (which is getter). In complex this adds a more congnitive load, than previous decison.

Reply:
You are correct. Thanks for catching this. I shall reject these comments and make better ones (since I did find some errors while trying this out myself).